osu.Game.Rulesets.Karaoke/Screens/Edit/Beatmaps/Singers/SingerScreen.cs

```diff
@@ -40,10 +43,8 @@ public SingerScreen()
     }
 
     [BackgroundDependencyLoader]
-    private void load(EditorBeatmap editorBeatmap, EditorClock editorClock)
+    private void load(EditorClock editorClock)
     {
-        BindablesUtils.Sync(selectedLyrics, editorBeatmap.SelectedHitObjects);
-
```

Comment:
Maybe use `schedule` instead?
e.g.
```
Schedule(() =>
{
    // todo: the reason why use schedule?
    BindablesUtils.Sync(selectedLyrics, editorBeatmap.SelectedHitObjects);
});
```

Reply:
The `InvalidThreadForMutationException` does recommend it, but based on the osu! codebase, bindable logic is generally done in `LoadComplete()` and it looks better and safer IMO.

Using either `Schedule` or `LoadComplete()` ensures it runs in the update thread, but `Schedule` should be used in more direct drawable mutation lines, see https://github.com/ppy/osu-framework/wiki/Threading#scheduling.

Reply:
OK
Seems make sense 👍🏼